### PR TITLE
No mpfr struct

### PR DESCRIPTION
**Pull request by ChrisPattison**
This removes the direct usage of the `__mpfr_struct` type in the tests. I am slightly worried that we might have been relying on some undefined behavior somewhere in the test by copying `__mpfr_struct`, so I removed the usage

### CODE REVIEW

File: host/Random.cpp [hunk 1]
```diff
@@ -9,32 +9,27 @@ RandomNumberGenerator::~RandomNumberGenerator() {
...
     mpfr_clear(mpfr_num);
     return num;
 }
 
-__mpf_struct RandomNumberGenerator::GenerateGmp() {
-    mpf_t num;
+void RandomNumberGenerator::GenerateGmp(mpf_ptr num) {
     mpf_init2(num, kMantissaBits);
-    Generate(num);
-    return num[0];
```
Comment (definelicht) [inline host/Random.cpp]: Isn't it missing the part where it actually generates anything? Right now it's doing nothing except initialization

File: host/Random.cpp [hunk 2]
```diff
@@ -9,32 +9,27 @@ RandomNumberGenerator::~RandomNumberGenerator() {
...
 }
 
-__mpfr_struct RandomNumberGenerator::GenerateMpfr() {
-    mpfr_t num;
+void RandomNumberGenerator::GenerateMpfr(mpfr_ptr num) {
     mpfr_init2(num, kMantissaBits);
-    Generate(num);
-    return num[0];
```
Comment (definelicht) [inline host/Random.cpp]: Isn't it missing the part where it actually generates anything? Right now it's doing nothing except initialization

**Review by definelicht — CHANGES_REQUESTED** [review]

**Comment by definelicht** [comment]
You are being very generous to their crappy interface design ;-) I wasn't worried about this, but I'm fine with it

File: host/Random.cpp (hunk 1 above)
Comment (ChrisPattison) [reply host/Random.cpp]: I changed some stuff at the last minute and apparently forgot to call generate....

**Pull request merged**